odoo/addons/base/models/ir_fields.py

```diff
@@ -42,6 +42,41 @@ def _format_import_error(self, error_type, error_msg, error_params=(), error_arg
                 error_params = tuple(sanitize(v) for v in error_params)
         return error_type(error_msg % error_params, error_args)
 
+    def _get_import_field_path(self, field, value):
+        """ Rebuild field path for import error attribution to the right field.
+        This method uses the 'parent_fields_hierarchy' context key built during treatment of one2many fields
+        (_str_to_one2many). As the field to import is the last of the chain (child_id/child_id2/field_to_import),
+        we need to retrieve the complete hierarchy in case of error in order to assign the error to the correct
+        column in the import UI.
+
+        :param (str) field: field in which the value will be imported.
+        :param (str or list) value:
+            - str: in most of the case the value we want to import into a field is a string (or a number).
+            - list: when importing into a one2may field, all the records to import are regrouped into a list of dict.
+                E.g.: creating multiple partners: [{None: 'ChildA_1', 'type': 'Private address'}, {None: 'ChildA_2', 'type': 'Private address'}]
+                where 'None' is the name. (because we can find a partner by his name, we don't need to specify the field.)
+
+        The field_path value is computed based on the last field in the chain.
+        for example,
+            - path_field for 'Private address' at childA_1 is ['partner_id', 'type']
+            - path_field for 'childA_1' is ['partner_id']
+
+        So, by retrieving the correct field_path for each value to import, if errors are raised for those fields,
+        we can the link the errors to the correct header-field couple in the import UI.
+        """
+        field_path = [field]
+        parent_fields_hierarchy = self._context.get('parent_fields_hierarchy')
+        if parent_fields_hierarchy:
+            field_path = parent_fields_hierarchy + field_path
+
+        field_path_value = value
+        while isinstance(field_path_value, list):
```

Comment:
I have to say that I don't understand that method at all ... why do we use value, why is it something a value, sometimes a list of dict ? I understand this is linked to relational fields but still, not understanding the computation here ...